app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -138,10 +145,15 @@ public void configure(Configuration configuration)
 
     logger.info("Fetching recent finished application runs between last time: " + (_lastTime + 1)
         + ", and current time: " + _currentTime);
+    String rmApplicationFilter = "";
+    if(elephantProperties.containsKey(RM_APPLICATION_FILTER)){
+      rmApplicationFilter = elephantProperties.getProperty(RM_APPLICATION_FILTER).replaceAll("^\"|\"$", "");
+    }
 
     // Fetch all succeeded apps
-    URL succeededAppsURL = new URL(new URL("http://" + _resourceManagerAddress), String.format(
-            "/ws/v1/cluster/apps?finalStatus=SUCCEEDED&finishedTimeBegin=%s&finishedTimeEnd=%s",
+    URL succeededAppsURL =
+        new URL(new URL("http://" + _resourceManagerAddress), String.format(
+            "/ws/v1/cluster/apps?finalStatus=SUCCEEDED&finishedTimeBegin=%s&finishedTimeEnd=%s&" + rmApplicationFilter,
```

Comment:
Instead of adding rmApplicationFilter , its better to add it as argument of String.format